indexing-service/src/main/java/org/apache/druid/indexing/overlord/config/WorkerTaskRunnerConfig.java

```diff
@@ -20,32 +20,66 @@
 package org.apache.druid.indexing.overlord.config;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.apache.druid.indexing.overlord.util.CustomJobTypeLimitsDeserializer;
+
+import java.util.HashMap;
+import java.util.Map;
 
 public class WorkerTaskRunnerConfig
 {
   @JsonProperty
   private String minWorkerVersion = "0";
 
   @JsonProperty
-  private double parallelIndexTaskSlotRatio = 1;
+  @JsonDeserialize(using = CustomJobTypeLimitsDeserializer.class)
+  private Map<String, Number> customJobTypeLimits = new HashMap<>();
```

Comment:
Why do we need a custom deserializer?
Can't we just have a map from String to Double?

Reply:
It was added mostly for validation as the idea is to have double / integer values to have an option to specify not only the ratio but also a limit